A github actions workflow to test if getting_started colab notebooks running properly in latest colab docker runtime env.

vkt1414 opened the pull request:
Prior to activating this workflow, we need to set three env variables:

SERVICE_ACCOUNT_KEY: Google Cloud service account key JSON (make sure to convert it to ONE LINE JSON). Note: minimum permissions required for the service account: Bigquery User
DOCKER_USERNAME: Docker Hub username.
DOCKER_PASSWORD: Docker Hub password or access token.

While this workflow is setup to test only the three getting_started notebooks, we can test any notebook that does not require a GPU. To accomplish that, every new notebook will need to be assessed if it requires any additional preprocessing steps.

vkt1414 commented:
A service account is now dedicated for this. Whenever we are ready to merge this PR, I can provide the service account key to create the github repo secret.

fedorov commented on test/outputs/output_part1_prerequisites.ipynb:
@vkt1414 let's inject a header that this is an artifact produced in the process of testing, and also rename those notebooks to include suffix "_papermill_output" in the name. I think this is important to clearly distinguish these notebooks from those users should use for the tutorial purposes.

vkt1414 commented:
@fedorov I modified the preprocessingNotebooks.py script to let the users know whenever we inject something. I also modified the papermill output notebook names. So I believe we are good to go.
Could you use your dockerhub credentials while creating these?

DOCKER_USERNAME: Docker Hub username.
DOCKER_PASSWORD: Docker Hub password or access token.
SERVICE_ACCOUNT_KEY
